### PR TITLE
Mccalluc/gcms

### PR DESCRIPTION
**Pull request by mccalluc**
Preview of the documentation is [here](https://github.com/hubmapconsortium/ingest-validation-tools/blob/700e3af49fe8b30fc51a9c08dace36526d779e19/docs/gcms/index.md).

Does this use an existing directory schema? If not, is it on @icaoberg's radar?

I'll call out a few places below where I wasn't sure about the right value. 

@ilan-gold : Don't need to look at the schema, but there are some code changes because there isn't a version 0; Please review.

### CODE REVIEW

File: src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml
Comment (mccalluc) [inline src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:74]: @cebriggs7135 : Anything besides cm? (Units must be enumerated.)

File: src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml
Comment (mccalluc) [inline src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:81]: Ditto.

File: src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml
Comment (mccalluc) [inline src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:108]: @cebriggs7135 : From the docs, guessing this isn't enumerated?

**Comment by ilan-gold** [comment]
Why is there no verison 0?

**Comment by mccalluc** [comment]
@ilan-gold 
> Why is there no version 0?

"version 0" are the schemas from the time before we realized we needed versioning. (They don't have a `version` column.) Any new schemas created now will start at version 1.

File: docs/field-descriptions.yaml
Comment (ilan-gold) [inline docs/field-descriptions.yaml:103]: So it this a boolean or a value?  Both?   Not so familiar with these docs but if there is some sort of constraint, it's not super obvious.

File: docs/field-descriptions.yaml
Comment (ilan-gold) [inline docs/field-descriptions.yaml:101]: Same.

File: docs/field-descriptions.yaml
Comment (ilan-gold) [reply docs/field-descriptions.yaml:103]: I can see from below that this is not a boolean so take this comment or leave it.

**Review by ilan-gold — APPROVED** [review]

File: src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml
Comment (mccalluc) [inline src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:23]: @cebriggs7135 -- In general, for enums, I'd prefer to avoid duplicating the enum in the description.

**Comment by cebriggs7135** [comment]
Directory structure to be determined. No GCMS data yet.

File: src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml
Comment (cebriggs7135) [reply src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:81]: answered above.

File: docs/field-descriptions.yaml
Comment (cebriggs7135) [inline docs/field-descriptions.yaml:60]:
```suggestion
column_temp_unit: Units for GC column temperature
    enum: Celsius
```

File: src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:81]:
```suggestion
      - Celsius
```

File: src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:74]:
```suggestion
      - meters
```

File: src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:23]:
```suggestion
  description: The ion source type used for surface sampling.
```

**Review by cebriggs7135 — COMMENTED** [review]
Column length units are usually meters.
Waiting for confirmation. 

Column temperature units are Celsius. That's confirmed by jeannie Camarillo.

File: src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml
Comment (mccalluc) [reply src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:74]: Do you want to correct the description? Also, we've typically just been using abbreviations -- That should hold here, as well?

File: src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml
Comment (mccalluc) [reply src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:81]: Typically, we've used abbreviations?

File: docs/field-descriptions.yaml
Comment (mccalluc) [reply docs/field-descriptions.yaml:60]: Handled below

**Comment by mccalluc** [comment]
@cebriggs7135 : for context, these are the other units across all schemas, and their counts:
```
src/generate_field_enum_csv.py | grep unit | perl -pne 's/.*,//' | sort | uniq -c
  13
   3 C
   4 Hz
   6 cm
   2 mL/min
   2 minutes
  10 mm
   2 nL/min
   2 nM
  14 ng
   2 ng/mg
  56 nm
   4 ug
   1 ug/ml
  68 um
```
- 'C' is already used three times.
- SI abbreviations are used consistently.

**Comment by mccalluc** [comment]
@cebriggs7135 -- Any updates?

**Comment by cebriggs7135** [comment]
Just sent a list of questions to the mass spec team AGAIN. These are re: 
ms_source for GC & CE- MS
column length units for GCMS
temp units for GCMS

**Comment by mccalluc** [comment]
@cebriggs7135 -- As discussed, we will merge this when the fields are stable... but it sounds like you still may need to add a column? Please clarify.

[in comments](https://github.com/hubmapconsortium/ingest-validation-tools/pull/853#discussion_r640661011):

> I'm waiting for confirmation that column length is in meters. I'm also asking about column_temperature units so we should add a column for that if you don't already have one.

**Comment by mccalluc** [comment]
@cebriggs7135 - I've changed this back to "draft" to put it on the back-burner. When all the required information has been gathered, ping me.

File: src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:30]:
```suggestion
      - nESI
      - EI
```

**Review by cebriggs7135 — CHANGES_REQUESTED** [review]

File: src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml
Comment (cebriggs7135) [reply src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:30]: per Elizabeth Neumann

File: src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:6]:
```suggestion
      - mass_spectrometry
```

Comment (cebriggs7135) [reply src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:6]: GC & CZE mass spec are not expected to be spatial assays but may be. Therefore my recommendation is to make the assay_categories ```mass_spectrometry``` and make the spatial fields ```required: FALSE```.

File: src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:14]:
```suggestion
```

Comment (cebriggs7135) [reply src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:14]: protein is not listed in the schema for GC MS.

File: src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:74]:
```suggestion
      - cm
```

Comment (cebriggs7135) [reply src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:74]: Column length for CE is cm and no one seems to respond to GC column length so cm is the safe option here.

File: src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml
Comment (cebriggs7135) [reply src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:81]: OK. I changed to C.

File: src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml
Comment (cebriggs7135) [inline src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:108]:
```suggestion
```

Comment (cebriggs7135) [reply src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:108]: No enums provided so this can be flexible.

**Review by cebriggs7135 — COMMENTED** [review]
I've just reviewed the GC-MS and responded to the remaining TO DOs .

File: src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml
Comment (mccalluc) [inline src/ingest_validation_tools/table-schemas/assays/gcms-v1.yaml:71]:
```suggestion
  description: Unit for GC column length
```

**Pull request merged**
